src/vertex/query.rs

```diff
@@ -0,0 +1,47 @@
+use crate::vertex::Vertex;
+use crate::QueryAttribute;
+
+impl Vertex {
+    /// Checks if "from" or "to" edge has an attribute
+    pub fn has_edge_attr(&self, attr_k: &str) -> bool {
+        self.get_from_edge().has_attr(attr_k) || self.get_to_edge().has_attr(attr_k)
+    }
+
+    /// Checks if "from" or "to" edge has a like attribute
+    pub fn like_edge_attr(&self, attr_k: &str) -> bool {
+        self.get_from_edge().like_attr(attr_k) || self.get_to_edge().like_attr(attr_k)
+    }
+
+    /// Checks if "from" or "to" edge has an attribute and equal for value
+    pub fn equals_edge_attr<T>(&self, attr_k: &str, attr_v: T) -> bool
+    where
+        T: std::fmt::Display + std::clone::Clone,
+    {
+        self.get_from_edge().equals_attr(attr_k, attr_v.clone()) || self.get_to_edge().equals_attr(attr_k, attr_v.clone())
+    }
+}
```

Comment:
Rename the implemented methods.
- `has_edge_attr` -> `has_edge_with_attr_key`
- `like_edge_attr` -> `has_edge_with_attr_key_like`
- `equals_edge_attr` -> `has_edge_with_attr_value_equal`